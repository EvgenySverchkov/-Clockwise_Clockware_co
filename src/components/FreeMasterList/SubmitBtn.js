import React from "react";
import PropTypes from "prop-types";

const SubmitBtn = ({ loading }) => (
  <div className="form-group float-right mt-3">
    <input
      type="submit"
      value={loading ? "Loading..." : "Book now"}
      className="btn btn-primary"
    />
  </div>
);

SubmitBtn.propTypes = {
  loading: PropTypes.bool,
};

export default SubmitBtn;
