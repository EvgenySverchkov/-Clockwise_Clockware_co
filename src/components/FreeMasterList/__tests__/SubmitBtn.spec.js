import React from "react";
import {create} from "react-test-renderer";

import SubmitBtn from "../SubmitBtn";

it("Test of <SubmitBtn/> for <FreeMasterList/>", ()=>{
    let mockProps = {
        loading: false
    }
    let component = create(<SubmitBtn {...mockProps}/>);

    expect(component.toJSON()).toMatchSnapshot();

    mockProps = {
        loading: true
    }
    component.update(<SubmitBtn {...mockProps}/>);

    expect(component.root.findByProps({type: "submit"}).props.value).toBe("Loading...");
});