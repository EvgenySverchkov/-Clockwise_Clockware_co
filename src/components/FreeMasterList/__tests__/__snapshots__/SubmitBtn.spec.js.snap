// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <SubmitBtn/> for <FreeMasterList/> 1`] = `
<div
  className="form-group float-right mt-3"
>
  <input
    className="btn btn-primary"
    type="submit"
    value="Book now"
  />
</div>
`;
